include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -88,6 +88,14 @@ namespace dht
 		virtual size_t num_peers() const = 0;
 #endif
 
+		// PENDING REVIEW
+		// This method notifies the list of all node's ids of each
+		// DHT running inside libtorrent. It's advisable that the
+		// concrete implementation keeps a copy of this list for an
+		// eventual prioritization when deleting an element to make
+		// room for a new one.
+		virtual void update_node_ids(std::vector<sha1_hash> const& ids) = 0;
```

Comment:
despite the fact that sha1_hash is the same type as node_id, I think there's some value in making the distinction. The dht namespace should have a typedef called "node_id"